[26.04] Docs: ROCm-RAG update (#9)
ROCm/rocm-rag: docs/install/configure-framework.rst

  :description: Set up and configure a ROCm-RAG framework
  :keywords: RAG, ROCm, install, Docker, frameworks, LLM

*************************************************************
ROCm-RAG installation
*************************************************************

This topic covers setup and install instructions to help you get started running ROCm-RAG.

System requirements
=============================================================

> To use ROCm-RAG `1.0.0 <https://github.com/ROCm/rocm-rag/tree/release/1.0.0>`__, you need the following prerequisites:

Reviewer: Do we need the version number here?

> - **ROCm version:** `6.4.1 <https://rocm.docs.amd.com/en/docs-6.4.1/>`__

Reviewer: Seems like an old version

- **Operating system:** Ubuntu 22.04
- **GPU platform:** AMD Instinct™ MI300X

.. note::

   - If you're hosting the LLM outside the Docker container, the container requires one MI300X GPU. By default, the GPU ID is ``0``, but this can be changed by setting the environment variables ``ROCM_RAG_EMBEDDER_TP`` and ``ROCM_RAG_EMBEDDER_GPU_IDS``.
   - If you're hosting the LLM inside the container (``ROCM_RAG_USE_EXAMPLE_LLM`` is set to ``true``), three MI300X GPUs are required. By default, the GPU IDs are ``0``, ``1``, ``2``, but these can be changed by setting the environment variables ``ROCM_RAG_EMBEDDER_TP``, ``ROCM_RAG_EMBEDDER_GPU_IDS``, ``ROCM_RAG_LLM_TP``, and ``ROCM_RAG_LLM_GPU_IDS``.

Install ROCm-RAG
==============================================================

To install ROCm-RAG on ROCm, you have the following options:

* :ref:`using-docker-with-rag-pre-installed` **(recommended)**
* :ref:`build-rocm-rag-docker-image`

> After setting up the container with either option, configure your RAG framework, inferencing framework, and environment variables before running the pipelines.

Reviewer: Should environment variables be linked to this?
https://rocm.docs.amd.com/projects/rocm-rag/en/docs-26.04/install/configure-framework.html#environment-variable-reference

.. _using-docker-with-rag-pre-installed:

Use a prebuilt Docker image with ROCm-RAG pre-installed
---------------------------------------------------------------

> The prebuilt image contains a fully configured ROCm-RAG installation and all required dependencies pre-installed.

Reviewer: ```suggestion
The prebuilt image contains a fully configured ROCm-RAG installation and all required dependencies.
```

> 1. Pull the Docker image.

Reviewer 2: Recommend removing the step number since there's only one step

   .. code-block:: bash 

      docker pull rocm/rocm-rag:rocm-rag-1.0.0-rocm6.4.1-ubuntu22.04

.. _build-rocm-rag-docker-image:

Build from source
---------------------------------------------------------------

ROCm-RAG can be built from source using the provided Dockerfile.

1. Clone the `https://github.com/ROCm/rocm-rag <https://github.com/ROCm/rocm-rag>`__ repository.

   .. code-block:: bash

      git clone https://github.com/ROCm/rocm-rag.git --recursive
      cd rocm-rag
  
2. Build the Docker image.
  
   .. code-block:: bash  
    
      docker build -t rocm-rag -f Dockerfile .

Configure ROCm-RAG
==============================================================

> Before running ROCm-RAG, you need to configure the RAG framework, inferencing framework, and environment variables.

Reviewer: Same env  var comment as above

            --host 0.0.0.0 --port 30000

After setting up your inference server, ensure you set the correct API endpoints for LLM server-related environment variables.

Configure environment variables
---------------------------------------------------------------

You can configure both extraction and retrieval parameters by setting environment variables for the Docker container.
There are three ways to set environment variables:

.. tab-set::

   .. tab-item:: .env file (recommended)

      1. Start with `default.env <https://github.com/ROCm/rocm-rag/blob/main/default.env>`__ as a base.
      2. Modify the variables as needed and provide the ``.env`` file when running the container:

         .. code-block:: bash

            docker run --env-file <your env file> ...

   .. tab-item:: Docker run

      Set variables individually when starting the container:

      .. code-block:: bash

         docker run -e VAR1=value1 -e VAR2=value2 ...

   .. tab-item:: Export in container

      Export variables inside the container when running in interactive mode:

      .. code-block:: bash

         export VAR1=value1
         export VAR2=value2

> Environment variable reference

Reviewer 2: Consider moving these variables to a separate reference article. Listing all the variables here is likely out of scope for the installation article

   * - ``ROCM_RAG_REQUIRE_HUMAN_VERIFICATION_FILTERS``
     - List of regex filters for identifying pages that require human verification
   * - ``ROCM_RAG_PAGE_NOT_FOUND_FILTERS``
     - List of regex filters for identifying not found pages
   * - ``ROCM_RAG_SET_MAX_NUM_PAGES``
     - Enable limit on the maximum number of pages to scrape
   * - ``ROCM_RAG_MAX_NUM_PAGES``
     - Maximum number of pages to scrape
   * - ``ROCM_RAG_MAX_CHUNK_LENGTH``
     - Maximum number of tokens for SemanticChunkMerger
   * - ``ROCM_RAG_SIMILARITY_THRESHOLD``
     - Similarity threshold for SemanticChunkMerger to merge

**Retrieval parameters**

.. list-table::
   :header-rows: 1
   :widths: 40 60

   * - Variable
     - Description
   * - ``ROCM_RAG_RETRIEVAL_FRAMEWORK``
     - Retrieval RAG framework (``haystack`` or ``langgraph``)
   * - ``ROCM_RAG_USE_EXAMPLE_LLM``
     - Deploy example LLM inference server inside this Docker
   * - ``ROCM_RAG_LLM_API_BASE_URL``
     - LLM API base URL
   * - ``ROCM_RAG_LLM_API_PORT``
     - LLM API port
   * - ``ROCM_RAG_LLM_MODEL``
     - LLM model
   * - ``ROCM_RAG_LLM_TP``
     - Tensor parallelism
   * - ``ROCM_RAG_LLM_GPU_IDS``

>      - Visible GPUs for example LLM

Reviewer: ```suggestion
     - Visible GPUs, for example, LLM
```

   * - ``ROCM_RAG_HAYSTACK_CERTAINTY_THRESHOLD``
     - Certainty threshold for retrieval
   * - ``ROCM_RAG_HAYSTACK_TOP_K_RANKING``
     - Top K retrieved documents for Haystack retrieval pipeline
   * - ``ROCM_RAG_LANGGRAPH_TOP_K_RANKING``
     - Top K retrieved documents for LangGraph retrieval pipeline

Run ROCm-RAG
==============================================================

> Now that the ROCm-RAG framework is configured, you can execute the extraction and retrieval pipelines through:

Reviewer: ```suggestion
Once the ROCm-RAG framework is configured, you can execute the extraction and retrieval pipelines through:
```

* :doc:`An interactive session <../how-to/run-interactive-session>`
* :doc:`Direct execution with your terminal <../how-to/direct-execute>`

Next steps
---------------------------------------------------------------

Now that you have ROCm-RAG configured on your AMD Instinct GPU, you can:

> 

Reviewer: Are these supposed to be ordered steps below? Just confirming as the section title is "next steps"
